pynet/models/spherical/sampling.py

```diff
@@ -273,6 +274,20 @@ def downsample(vertices, target_vertices):
     return nearest_idx.squeeze()
 
 
+def downsample_data(data, down_indices, neighs):
+    if len(data.shape) < 3:
+        data = data[np.newaxis, :, :]
+    data = data.transpose((0, 2, 1))
+    for i in range(len(down_indices)):
+        down_neigh_indices = neighs[i][down_indices[i]]
+        n_vertices, neigh_size = down_neigh_indices.shape
+
+        data = data[:, :, down_neigh_indices].reshape(
+                len(data), data.shape[1], n_vertices, neigh_size)
+        data = data.mean(-1)
```

Comment:
If you are using an icosahedron you have a perfect match when you downsampled (cf. downsample function).
Why do you need to take the mean here?